kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -423,41 +422,29 @@ public void deactivateSubscription(Subscription subscription, DeactivationReason
     }
 
     @Override
+    @Cacheable("pack-type")
```

Comment:
Just to expand on this, what we are doing with this cache is creating a shortcut to find an instance of an object.  That object can be identified by many different "keys".  If we split our cache by key then we eliminate the caches ability to better manage the objects.  The cache internals may only maintain an object once even though multiple keys point to it.  However if we have many caches for the same object we prevent it from doing that for us.  (if the cache implementation actually does that)